src/Components/ExperimentOne/ExperimentOneUserForm.js

```diff
@@ -39,15 +14,16 @@ function ExperimentOneUserForm(props) {
     setNumberOfParticipants(event.target.value);
     setIsSubmitPressed(false);
   };
+
   const handleSubmit = (event) => {
     event.preventDefault();
-    let randomActivity =
-      activityTypes[Math.floor(Math.random() * activityTypes.length)];
-    setUserActivity(randomActivity);
     setIsSubmitPressed(true);
-    // alert("The user is called "+userName+" and number of friends free today are "+numberOfParticipants+
-    // ". Based on your personality, I would suggest doing something "+randomActivity);
   };
+
+  useEffect(() => {
```

Comment:
This feels wrong... why not just call this function inside the onSubmit handler?

Reply:
Update on this
As we discussed over Slack, I did pass a `true` value in the handleSubmitButton as follows:
`  const handleSubmit = (event) => {`
    `event.preventDefault();`
    `setIsSubmitPressed(true);`
  ` props.userInfoFormData(userName, numberOfParticipants, [true);};`

However since I am passing an exact boolean value, so it is always set to true and so I had to add another statement in other handles to reset this value to `false` as follows:
`  const handleUserNameChange = (event) => {`
    `setUserName(event.target.value);`
    `setIsSubmitPressed(false);`
    `props.userInfoFormData(userName, numberOfParticipants, false);};`

`  const handleNumberOfParticipants = (event) => {`
   ` setNumberOfParticipants(event.target.value);`
   ` setIsSubmitPressed(false);`
    `props.userInfoFormData(userName, numberOfParticipants, false);};`

Do you think there is a better way to refactor this or you think this is acceptable? 🤔

Reply:
I think this is now done and looks better. Thank you for your help with this. I will push the changes now